### ISSUE TITLE
Unpractical type casting for Vector Constraints

### ISSUE THREAD

**Issue by PierreTsr** [corail-research/SeaPearl.jl#118]
Currently all the constraints that take a vector of variables as input are casted as `Vector{AbstractIntVar}`. This is useful in order to define constraints on `IntVar` but also on `IntVarView`.
However, by doing it that way, the user is forced to cast its vector as `AbstractIntVar` eventhough the vector contains only a more specific variable type, which is quite unpractical.

This could be fixed easily by casting the input of these constraints as `Vector{<:AbstractIntVar}`, and it would even allow a more specific dispatch in the constraints if needed.

**Comment by qcappart** [comment]
I agree , it seems to be a suited improvement. @ilancoulon and @felixchalumeau do you see any issue with this ?

**Comment by ilancoulon** [comment]
I think it is a a great idea!
